tests/unit/config/interpolation_test.py

```diff
@@ -146,3 +146,9 @@ def test_interpolate_missing_with_default(defaults_interpolator):
 def test_interpolate_with_empty_and_default_value(defaults_interpolator):
     assert defaults_interpolator("ok ${BAR:-def}") == "ok def"
     assert defaults_interpolator("ok ${BAR-def}") == "ok "
+
+
+def test_interpolate_with_chained_fallback_variables(defaults_interpolator):
+    assert defaults_interpolator("ok ${BAR:-$FOO}") == "ok first"
```

Comment:
Would this be a better format?

${BAR:-${FOO}}

Syntax for compose interpolation is consistent, otherwise it is unclear if we are expecting interpolation from compose or the shell.

Reply:
The simplest implementation that I've made throws "syntax error" while expanding nested curly braces. But I'll take a deeper look.

Reply:
Also, I should remember that compose **do** expands variables outside curly braces:

> Both $VARIABLE and ${VARIABLE} syntax are supported. Additionally when using the 2.1 file format, it is possible to provide inline default values using typical shell syntax:
https://docs.docker.com/compose/compose-file/#variable-substitution

And the same docs advise to use double-dollar when you want a single dollar:
> If you forget and use a single dollar sign ($), Compose interprets the value as an environment variable and will warn you:
>
> The VAR_NOT_INTERPOLATED_BY_COMPOSE is not set. Substituting an empty string.

So, there is no "shell expansion" inside docker-compose.yaml. Even ```$$variables``` are translated to literal ```$variables``` inside the containers, which is exactly what the [referred issue](#4836) states.

Reply:
Okay. As long as `the ${BAR:-$$FOO}` results in `the $FOO` and `the $BAR:-$FOO` results in `the first` (when $FOO has a value of `first`) then I am happy with what you have now.

Reply:
Would this cause problems in the following case:

`APIURL=${APP_APIURL:-http://$ENVIRONMENTsite.com/api}`

Where `ENVIRONMENT` is `dev` I would expect `APIURL` to come out as `http://devsite.com/api`